### PR TITLE
[Bug fixing][Eagle3] Always teacher-force TTT tokens; remove --use-off-policy-tokens

### PR DESCRIPTION
**Pull request by WindChimeRan**
## Summary

Eagle3 training simulates multi-step drafting ("training-time test", TTT). This PR removes the `--use-off-policy-tokens` flag and its default code path so that Eagle3 **always teacher-forces the ground-truth token** in the TTT loop.

## Mechanism

Each TTT step feeds the draft two things to predict the next token: the draft's **own hidden state**, carried forward from the previous step, and a **token embedding**. The hidden-state recursion — the part that mirrors what the draft does at inference — runs in every case and is untouched here.

The flag only chose the **token** source:

- old default: the draft's own argmax prediction;
- flag on: the ground-truth token from the data.

Where the draft predicts correctly the two are identical; they differ only at positions where it mispredicts.

## Why always teacher-force, and why deleting the alternative is safe

- **Consistent objective.** The training label at each position is the verifier's next-token distribution, computed over the **ground-truth prefix**. Teacher forcing feeds that same ground-truth prefix, so the draft's input and its label always agree. Feeding the draft's own token feeds a prefix the label was never computed for — an inconsistent target exactly where the draft is wrong.
- **Nothing lost at inference.** A mispredicted draft token is rejected by the verifier and truncates the draft chain, so training the draft to continue from its own mistake gains nothing at serving time.

So the removed path only ever added an inconsistent — and at best neutral — training target. Teacher forcing is the right default.

## Changes

- `eagle3/core.py`: teacher-force unconditionally in the TTT loop; drop the now-dead draft→verifier (`d2t`) remap that only applied to the self-token feed; remove the `use_off_policy_tokens` parameter.
- `eagle3/core.py`: drop it from the train/val trainer kwargs — validation now matches training.
- `scripts/train.py`, `docs/cli/train.md`: remove the CLI flag and its doc entry.

## Note

This changes the **default** training behavior (previously the draft's own tokens were fed by default). No config, checkpoint, or data-format changes. Scripts that passed `--use-off-policy-tokens` should drop the flag.

Verified: `tests/unit/train/test_cli_args.py` green (20 passed), module imports, ruff clean.

### CODE REVIEW

File: src/speculators/models/eagle3/core.py [hunk 1]
```diff
@@ -318,26 +317,21 @@ def forward(  # noqa: C901
             input_ids = torch.argmax(logits, dim=-1)
             draft_tokens.append(input_ids.detach().clone())
             # shape: [1, total_seq_len]
-            # Use d2t to map draft tokens to verifier tokens.
-            # Must be in verifier vocabulary space because we use the full verifier
-            # vocabulary in the embedding.
-            if self.d2t is not None:
```
Comment (shanjiaz) [inline src/speculators/models/eagle3/core.py:324]: why removing the d2t mapping?

Comment (WindChimeRan) [reply src/speculators/models/eagle3/core.py:324]: It's dead code. remap's output was overwritten on the next line

**Review by shanjiaz — APPROVED** [review]
